### PR TITLE
fix: unexpected commands and arguments should throw

### PR DESCRIPTION
**Pull request by threepointone**
This enables strict mode in our command line parser (yargs), so that unexpected commands and options uniformly throw errors.

Fixes https://github.com/cloudflare/wrangler2/issues/706

### CODE REVIEW

File: packages/wrangler/src/__tests__/index.test.ts [hunk 1]
```diff
@@ -854,7 +846,7 @@ describe("wrangler", () => {
               Try using \`wrangler dev\` to to try out a worker during development.
               "
             `);
-      await expect(runWrangler("preview GET 'Some Body'")).rejects
+      await expect(runWrangler(`preview GET "SomeBody"`)).rejects
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/index.test.ts:849]: Why this change?

**Review by petebacondarwin — APPROVED** [review]
Love it!

File: packages/wrangler/src/__tests__/index.test.ts (hunk 1 above)
Comment (threepointone) [reply packages/wrangler/src/__tests__/index.test.ts:849]: our faux parser splits on spaces naively so that's interpreted as two separate arguments https://github.com/cloudflare/wrangler2/blob/52ea60f2c0e082e7db5926cca74d79f48afbdf3b/packages/wrangler/src/__tests__/helpers/run-wrangler.ts#L9 I didn't want to shave that yak today and it wasn't affecting anything else, so decided to just simplify the test.
